### PR TITLE
Ensured address_pou is valid and corrected state parameters when building VAProfile::V3::Address objects

### PR DESCRIPTION
**Pull request by RachalCassity**
…request body

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

Bugs found during local testing. 1) The address_pou has to be either RESIDENCE or CORRESPONDENCE, not RESIDENCE/CHOICE. 2) Fixed state data not rendered in VAProfile::V3::Address model. 

Users with an ICN are verified users. Only verified users are permitted to use profile services. VAProfileRedis::V2::ContactInformation should only obtain data from verified users.

Removed logs that were added in this [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/19797).

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/98561

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/models/va_profile_redis/v2/contact_information.rb [hunk 1]
```diff
@@ -179,6 +179,10 @@ def response_from_redis_or_service
       def contact_info_service
         @service ||= VAProfile::V2::ContactInformation::Service.new @user
       end
+
+      def verified_user?
+        @user.icn.present?
```
Comment (RachalCassity) [inline app/models/va_profile_redis/v2/contact_information.rb:184]: User's with ICN are verified users. Only verified users are permitted to profile services.

**Review by ericboehs — APPROVED** [review]

File: app/models/va_profile_redis/v2/contact_information.rb [hunk 2]
```diff
@@ -179,6 +179,10 @@ def response_from_redis_or_service
       def contact_info_service
         @service ||= VAProfile::V2::ContactInformation::Service.new @user
       end
+
```
Comment (RachalCassity) [inline app/models/va_profile_redis/v2/contact_information.rb:182]: Users with an ICN are verified users. Only verified users are permitted to use profile services. VAProfileRedis::V2::ContactInformation should only obtain data from verified users.

File: app/models/user.rb [hunk 3]
```diff
@@ -415,6 +415,8 @@ def va_profile_email
   end
 
   def vaprofile_contact_info
+    return nil unless VAProfile::Configuration::SETTINGS.contact_information.enabled && icn.present?
```
Comment (RachalCassity) [inline app/models/user.rb:418]: Ensure icn is present before attempting to trying to retrieve Contact Info

File: app/models/va_profile_redis/v2/contact_information.rb (hunk 1 above)
Comment (LindseySaari) [inline app/models/va_profile_redis/v2/contact_information.rb:184]: What about the loa3 though? Does that not play a role anymore?

Comment (RachalCassity) [reply app/models/va_profile_redis/v2/contact_information.rb:184]: ICN is the only thing neccessary for the upgrade Contact Information service. ICN == verified users

**Review by LindseySaari — APPROVED** [review]

**Review by bramleyjl — APPROVED** [review]

**Pull request merged**